tokio/src/runtime/thread_pool/worker.rs

```diff
@@ -466,11 +478,17 @@ impl Context {
         *self.core.borrow_mut() = Some(core);
 
         // Park thread
+        if let Some(f) = &self.worker.shared.before_park {
+            f()
+        }
```

Comment:
This will also run when the runtime runs maintenance without actually parking (i.e. when duration is zero).